lib/timers.js

```diff
@@ -272,8 +309,36 @@ function setImmediate(callback, arg1, arg2, arg3) {
   return new Immediate(callback, args);
 }
 
-setImmediate[customPromisify] = function(value) {
-  return new Promise((resolve) => new Immediate(resolve, [value]));
+setImmediate[customPromisify] = function(value, options = {}) {
+  if (options == null || typeof options !== 'object') {
+    return Promise.reject(
+      new ERR_INVALID_ARG_TYPE(
+        'options',
+        'Object',
+        options));
+  }
+  const { signal } = options;
+  if (signal !== undefined &&
+    (signal === null ||
+     typeof signal !== 'object' ||
+     !('aborted' in signal))) {
+    return Promise.reject(
+      new ERR_INVALID_ARG_TYPE(
+        'options.signal',
+        'AbortSignal',
+        signal));
+  }
+  if (signal && signal.aborted)
+    return Promise.reject(lazyDOMException('AbortError'));
+  return new Promise((resolve, reject) => {
+    const immediate = new Immediate(resolve, [value]);
```

Comment:
```suggestion
  return new Promise((resolve, reject) => {
    if (signal?.aborted) {
      return reject(lazyDOMException('AbortError'));
    }
    const immediate = new Immediate(resolve, [value]);
```

Reply:
I want to have the option of eventually backporting this to 12.x if at all possible so would rather avoid the optional chaining.

Reply:
`signal && signal.aborted` then :P

Reply:
Also, given the other rejection cases for arg type validation, I'd rather keep the rejections consistent.

Reply:
Unless there's a strong objection here, I'm inclined to resolve this comment without action.